src/main/java/uk/gov/service/notify/NotificationClient.java

```diff
@@ -329,21 +357,38 @@ private HttpURLConnection getConnection(URL url) throws IOException {
         return conn;
     }
 
-    private JSONObject createBodyForPostRequest(final String templateId, final String phoneNumber, final String emailAddress, final Map<String, String> personalisation, final String reference) {
+    private JSONObject createBodyForPostRequest(final String templateId,
+                                                final String phoneNumber,
+                                                final String emailAddress,
+                                                final Map<String, String> personalisation,
+                                                final String reference,
+                                                final String encodedFileData) {
         JSONObject body = new JSONObject();
+
         if(phoneNumber != null && !phoneNumber.isEmpty()) {
             body.put("phone_number", phoneNumber);
         }
+
         if(emailAddress != null && !emailAddress.isEmpty()) {
             body.put("email_address", emailAddress);
         }
-        body.put("template_id", templateId);
+
+        if(templateId != null && !templateId.isEmpty()) {
```

Comment:
maybe we should get the org.apache.commons.lang3StringUtils.isBlank(). Unless we want to reduce the dependencies

Reply:
Done